src/core/dynamicalsystem_interface.jl

```diff
@@ -241,6 +242,18 @@ but a developer may care.
 """
 SciMLBase.isinplace(ds::DynamicalSystem) = errormsg(ds)
 
+"""
+    successful_step(ds::DynamicalSystem) -> true/false
+
+Return `true` if the last `step!` call to `ds` was successful, `false` otherwise.
+For continuous time systems this uses DifferentialEquations.jl error checking,
+for discrete time it checks if any variable is `Inf` or `NaN`.
+
+"""
+successful_step(ds::DynamicalSystem) = true
+successful_step(ds::DiscreteTimeDynamicalSystem) = all(x -> isfinite(x) , current_state(ds))
```

Comment:
You forgot the `!isnan` check here!

Reply:
Note that this is not `isinf`. I used `isfinite` which gives false for both `Inf`s and `NaN`s

Reply:
ok perfect you are right!